src/components/posts/video-player-overlay.tsx

```diff
@@ -0,0 +1,144 @@
+import Link from 'next/link'
+import Image from 'next/image'
+
+import {
+  useVideoPlayerOverlay,
+  type CompletedAction,
+} from '@/hooks/mux/use-video-player-overlay'
+import {Post} from '@/pages/[post]'
+import Spinner from '@/spinner'
+import {Button} from '@/ui/button'
+import {ArrowRight} from 'lucide-react'
+import {RefreshCw} from 'lucide-react'
+import {Card, CardContent, CardTitle, CardHeader, CardFooter} from '@/ui/card'
+import {motion} from 'framer-motion'
+
+type VideoPlayerOverlayProps = {
+  resource: Post
+}
+
+const VideoPlayerOverlay: React.FC<VideoPlayerOverlayProps> = ({resource}) => {
+  const {state: overlayState, dispatch} = useVideoPlayerOverlay()
+
+  switch (overlayState.action?.type) {
+    case 'COMPLETED':
+      const {playerRef, cta} = overlayState.action as CompletedAction
+
+      if (cta === 'cursor_workshop') {
+        return (
+          <motion.div
+            initial={{opacity: 0}}
+            animate={{opacity: 1}}
+            exit={{opacity: 0}}
+            aria-live="polite"
+            className="z-40 bg-background/85 absolute left-0 top-0 flex h-full w-full flex-col items-center justify-center pb-6 backdrop-blur-md sm:pb-16 text-white"
+          >
+            <CursorCTAOverlay
+              signUpLink="/workshop/cursor"
+              onReplay={() => {
+                if (playerRef.current) {
+                  playerRef.current.play()
+                }
+                dispatch({type: 'HIDDEN'})
+              }}
+            />
+          </motion.div>
+        )
+      }
+
+      return (
+        <motion.div
+          initial={{opacity: 0}}
+          animate={{opacity: 1}}
+          exit={{opacity: 0}}
+          aria-live="polite"
+          className="z-40 bg-background/85 absolute left-0 top-0 flex h-full w-full flex-col items-center justify-center pb-6 backdrop-blur-md sm:pb-16 text-white"
+        >
+          <div className="flex flex-col items-center justify-center gap-2 p-4">
+            <h2 className="text-lg sm:text-2xl font-bold mb-4 text-center">
+              {resource.fields.title}
+            </h2>
+            <Button
+              variant="outline"
+              onClick={() => {
+                if (playerRef.current) {
+                  playerRef.current.play()
+                }
+                dispatch({type: 'HIDDEN'})
+              }}
+              className="border border-blue-600 hover:bg-blue-600"
+            >
+              <RefreshCw className="mr-2 h-4 w-4" />
+              Watch again
+            </Button>
```

Comment:
default overlay is simple right now with a watch again button. @vojtaholik we could put vector search results here if we implemented that